File: Java/2 SECURITY FUNCTIONS/2 Improper Authorization/java/MOPAS_1_improper_authorization.java
/**
 *  MOPAS Improper Authorization Example 1
 *  Fetch private messages from MySQL database using DriverManager class
 *  Requirements: MySQL connector
 */
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.sql.*;

@WebServlet(urlPatterns = {"/MOPAS_1_improper_authorization"})
public class MOPAS_1_improper_authorization extends HttpServlet {

    @Override
    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws IOException, ServletException {

        // fake method only to demonstrate that action requires authentication
        if (!isAuthenticated()) {
            return;
        }

        // doesn't check that message with this id belong to current user
        String mid = request.getParameter("mid");
        String query = "SELECT * FROM private_message WHERE mid = ?";

        Connection connection = null;
        PreparedStatement statement = null;
        ResultSet resultSet = null;

        String driverName = "com.mysql.jdbc.Driver";

        try {
            Class.forName(driverName);

            connection = DriverManager.getConnection(
                    "jdbc:mysql://localhost:3306/db?user=user&password=pass"
            );
            statement = connection.prepareStatement(query);
            statement.setInt(1,Integer.valueOf(mid));
            resultSet = statement.executeQuery();

            while (resultSet.next()) {
                // do something
            }
        }
        catch (ClassNotFoundException e) {
        }
        catch (SQLException ex) {
        }
        finally {
            try {
                if (resultSet != null) {
                    resultSet.close();
                }
                if (statement != null) {
                    statement.close();
                }
                if (connection != null) {
                    connection.close();
                }
            }
            catch (SQLException ex) {
            }
        }
    }

    private boolean isAuthenticated() {
        return true;
    }
}

File: PHP/2 SECURITY FUNCTIONS/2 Improper Authorization/php/MOPAS_1_improper_authorization.php
<?php
/**
 *  MOPAS Improper Authorization Example 1
 *  Fetch private messages from MySQL database using old mysql extension
 */
define('DB_HOST', '');
define('DB_PORT', 3306);
define('DB_USER', '');
define('DB_PASSWORD', '');
define('DB_NAME', '');

// fake method only to demonstrate that action requires authentication
if (!IsAuthenticated()) {
	die('access denied');
}

if (empty($_GET['mid'])) {
	die('no message id');
}

$mid = (int)$_GET['mid'];

error_reporting(0);

if (mysql_connect(DB_HOST . ':' . DB_PORT, DB_USER, DB_PASSWORD) === FALSE) {
	die('Could not connect to database');
}

if (mysql_select_db(DB_NAME) === FALSE) {
	die('Could not select database');
}

if (mysql_set_charset('utf8') === FALSE) {
	die('Unable to set the character set');
}

// no check that message with this id belong to current user
$query = "SELECT * FROM private_message WHERE mid = '$mid'";

$result = mysql_query($query);

if ($result === FALSE) {
	die('Query error.');
}

$numRows = mysql_num_rows($result);

if ($numRows === FALSE || $numRows === 0) {
	die('No message for you with this id');
}

while($messageInfo = mysql_fetch_assoc($result)) {
	// do anything
}

function IsAuthenticated() {
	return true;
}
?>